packages/core/src/execution-engine/node-execution-context/node-execution-context.ts

```diff
@@ -178,6 +180,22 @@ export abstract class NodeExecutionContext implements Omit<FunctionsBase, 'getCr
 		return this.instanceSettings.instanceId;
 	}
 
+	protected getExecutionWaitingToken() {
+		const token = crypto
+			.createHmac('sha256', this.instanceSettings.encryptionKey)
+			.update(this.getExecutionId())
```

Comment:
Signing only the execution id is **not** secure. It allows changing any query param and the signature will still be valid. It also makes any subsequent wait URL have the same signature. Instead we need to sugn all critical parts, at least path and all query params.

Reply:
we could not include query params, as we expect them be different, depending of users choice
I think we could add stringified node's parameters to update, this would be unique per waiting node and we could decode it later, WDYT?

Reply:
Oh so users can provide arbitrary input to the hitl nodes using query params? I was thinking of signing the URL but I guess we could also sign the details that identify the 1) execution 2) node 3) input (unless arbitrary)

Reply:
currently it's execution id  + stringified node(INode), I think this would be unique and secure